tests/config/test_config.py

```diff
@@ -20,7 +21,7 @@
 
 
 def check_configuration_file_reader(
-    runner: Run,
+    runner: Run | LintRun,
```

Comment:
```suggestion
    runner: Run,
```

Should be fine since `_Run` subclasses `Run`.

Reply:
We need a typevar for this to work, right ?

Reply:
I think it should be fine? But I might be mistaken. It is the same as using `_ArgumentsProvider` when we pass `BaseChecker`.

Reply:
I started with Run only the error is:
```
tests/config/test_config.py:56: error: Argument 1 to "check_configuration_file_reader" has incompatible type "Run"; expected "_Run"  [arg-type]
Found 1 error in 1 file (checked 256 source files)
```

Reply:
Ah, it should be the other way around. It should be typed as `LintRun` (the `pylint.lint` one). That should accept both.

Reply:
Damn that was a stupid mistake, my bad.

Reply:
5000538894d9fbc113c04bf5e5f8152de72ba25d